### PR TITLE
Solve the assignment

### PR DESCRIPTION
**Pull request by ReemG**

### CODE REVIEW

File: assignments/assignment11/company.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+var Manager = function(name,department,reports){
+    this.name = name; 
+    this.department = department;
+    this.role = "manager";
+    this.reports = reports;
+}
+Manager.prototype.getInfo = function(){
+    return Employee.prototype.getInfo.call(this)+ " I manage " + this.reports + " employees." ; 
+}
+Manager.prototype.__proto__ = Employee.prototype;
```
Comment (AhmadBaracat) [inline assignments/assignment11/company.js:19]: We don't want to use `__proto__`

File: assignments/assignment11/task-manager.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        return tasks;
+    }
+    find = function (matcher) {
+        var arr = [];
+        for (var i = 0; i < tasks.length; i++) {
+            var t = true;
+            var c = true;
+            var lowertitle = tasks[i].title.toLowerCase();
+            var lowercategory = tasks[i].category.toLowerCase();
+            for (var j = 0; j < matcher.length; j++) {
```
Comment (AhmadBaracat) [inline assignments/assignment11/task-manager.js:28]: You don't have to write the matching algorithm yourself. Use the builtin methods.

File: assignments/assignment11/task.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+var Task = function (category, title, priority, estimate) {
+    this.category = category;
+    this.title = title;
+    this.priority = priority;
+    this.estimate = estimate;
+    this.spent = 0;
+    this.remaining = 1.5;
```
Comment (AhmadBaracat) [inline assignments/assignment11/task.js:7]: This should not be hardcoded :)

File: assignments/assignment11/task.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    if (value >= 0) {
+        this.spent = value;
+        if (this.estimate - value > 0)
+        { this.remaining = this.estimate - value; }
+        else { this.remaining = 0; }
+    }
+
+    else if (value < 0 || typeof value !== 'number') {
+        this.spent = 0;
+        this.remaining = 1.5;
```
Comment (AhmadBaracat) [inline assignments/assignment11/task.js:19]: This shouldn't be hardcoded :)

File: assignments/assignment11/task.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+}
+Task.prototype.done = function () {
+    return this.remaining === 0
+}
+Task.prototype.complete = function () {
+    this.remaining = 0;
+    this.track(1.5);
```
Comment (AhmadBaracat) [inline assignments/assignment11/task.js:28]: Here too...

**Review by AhmadBaracat — APPROVED** [review]

File: assignments/assignment11/task.js (hunk 3 above)
Comment (ReemG) [reply assignments/assignment11/task.js:7]: should it be random ? or what ?

Comment (AhmadBaracat) [reply assignments/assignment11/task.js:7]: Should be calculated from the parameters passed to the function.

**Pull request merged**